File: software/examples/python/example_motor.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

HOST = "localhost"
PORT = 4223
UID = "XYZ" # Change XYZ to the UID of your Motorized Linear Poti Bricklet

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_motorized_linear_poti import BrickletMotorizedLinearPoti

# Callback function for position reached callback
def cb_position_reached(position):
    print("Position: " + str(position)) # Range: 0 to 100

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    mlp = BrickletMotorizedLinearPoti(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected

    # Register position reached callback to function cb_position_reached
    mlp.register_callback(mlp.CALLBACK_POSITION_REACHED, cb_position_reached)

    # Move slider smooth to the middle
    mlp.set_motor_position(50, mlp.DRIVE_MODE_SMOOTH, False)

    raw_input("Press key to exit\n") # Use input() in Python 3
    ipcon.disconnect()

File: software/examples/php/ExampleMotor.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletMotorizedLinearPoti.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletMotorizedLinearPoti;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your Motorized Linear Poti Bricklet

// Callback function for position reached callback
function cb_positionReached($position)
{
    echo "Position: $position\n"; // Range: 0 to 100
}

$ipcon = new IPConnection(); // Create IP connection
$mlp = new BrickletMotorizedLinearPoti(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Register position reached callback to function cb_positionReached
$mlp->registerCallback(BrickletMotorizedLinearPoti::CALLBACK_POSITION_REACHED,
                       'cb_positionReached');

// Move slider smooth to the middle
$mlp->setMotorPosition(50, BrickletMotorizedLinearPoti::DRIVE_MODE_SMOOTH, FALSE);

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>
